src/ImageSharp/Memory/Allocators/ArrayPoolMemoryAllocator.cs

```diff
@@ -12,140 +13,131 @@ namespace SixLabors.ImageSharp.Memory
     /// </summary>
     public sealed partial class ArrayPoolMemoryAllocator : MemoryAllocator
     {
-        private readonly int maxArraysPerBucketNormalPool;
-
-        private readonly int maxArraysPerBucketLargePool;
+        /// <summary>
+        /// The upper threshold to pool arrays the shared buffer. 1MB
+        /// This matches the upper pooling length of <see cref="ArrayPool{T}.Shared"/>.
+        /// </summary>
+        private const int SharedPoolThresholdInBytes = 1024 * 1024;
 
         /// <summary>
-        /// The <see cref="ArrayPool{T}"/> for small-to-medium buffers which is not kept clean.
+        /// The default value for the maximum size of pooled arrays in bytes. 2MB.
         /// </summary>
-        private ArrayPool<byte> normalArrayPool;
+        internal const int DefaultMaxArrayLengthInBytes = 2 * SharedPoolThresholdInBytes;
```

Comment:
This is also a parameter we can tweak!

Reply:
Yep. Time is against me now though, it's nearly 3am.